src/t_hash.c

```diff
@@ -2704,16 +2704,21 @@ static ExpireMeta *hashGetExpireMeta(const eItem hash) {
         serverPanic("Unknown encoding: %d", hashObj->encoding);
     }
 }
-
+/* HTTL key <FIELDS count field [field ...]>  */
 static void httlGenericCommand(client *c, const char *cmd, long long basetime, int unit) {
     UNUSED(cmd);
     robj *hashObj;
-    long numFields = 0, numFieldsAt = 2;
+    long numFields = 0, numFieldsAt = 3;
 
     /* Read the hash object */
     if ((hashObj = lookupKeyReadOrReply(c, c->argv[1], shared.null[c->resp])) == NULL ||
         checkType(c, hashObj, OBJ_HASH)) return;
 
+    if (strcasecmp(c->argv[numFieldsAt-1]->ptr, "FIELDS")) {
+        addReplyError(c, "Constant argument FIELDS is missing or not at the right position");
+        return;
+    }
+
     /* Read number of fields */
     if (getRangeLongFromObjectOrReply(c, c->argv[numFieldsAt], 1, LONG_MAX,
                                       &numFields, "Parameter `numFileds` should be greater than 0") != C_OK)
```

Comment:
maybe as part of this pr, you can fix the error message. There is typo in 'numFields'. Also, maybe you can say "invalid number of fields" instead of "numFields" in the error message. 

There are more cases like that in the file, fyi.